src/nvim/option.c

```diff
@@ -301,6 +301,15 @@ static char *(p_cot_values[]) = {"menu", "menuone", "longest", "preview",
 # include "option.c.generated.h"
 #endif
 
+static void set_runtimepath_default(void)
+{
+  garray_T rtp_ga;
+  ga_init(&rtp_ga, (int)sizeof(const char *), 1);
+  GA_APPEND(const char *, &rtp_ga, get_user_conf_dir());
+  GA_APPEND(const char *, &rtp_ga, concat_fnames(get_user_conf_dir(), "after", true));
+  set_string_default("runtimepath", ga_concat_strings(&rtp_ga));
```

Comment:
The safest bet for allocating path names is just `xmalloc(PATH_MAX)`. If you care about over-allocated memory, you can `xrealloc(rtp strlen(rtp) + 1)` later, but it's not necessary.

I just realized another issue: It looks like the resulting string will be `"{get_user_conf_dir()}{get_user_conf_dir()}/after"`. Could just do this:
```c
char* conf_dir = get_user_conf_dir();
set_string_default("runtimepath", concat_fnames(conf_dir, "after", true);
xfree(conf_dir);
```